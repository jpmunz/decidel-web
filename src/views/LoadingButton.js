import React from "react";
import { Button, Spinner } from "react-bootstrap";

class LoadingButton extends React.Component {
  render() {
    const { loading, text, ...otherProps } = this.props;

    if (loading) {
      return (
        <Button disabled {...otherProps}>
          <Spinner
            as="span"
            animation="border"
            size="sm"
            role="status"
            aria-hidden="true"
          />
          <span className="sr-only">Loading...</span>
        </Button>
      );
    } else {
      return <Button {...otherProps}>{text}</Button>;
    }
  }
}

export default LoadingButton;
