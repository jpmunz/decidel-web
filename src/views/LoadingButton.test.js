import React from "react";
import LoadingButton from "./LoadingButton";
import { render } from "@testing-library/react";

test("Render loading state", () => {
  expect(
    render(
      <LoadingButton loading={true} text="Action" variant="primary" />
    ).asFragment()
  ).toMatchSnapshot();
});

test("Render default state", () => {
  expect(
    render(
      <LoadingButton loading={false} text="Action" variant="primary" />
    ).asFragment()
  ).toMatchSnapshot();
});
