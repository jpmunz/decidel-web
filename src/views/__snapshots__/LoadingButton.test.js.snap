// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default state 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
    type="button"
  >
    Action
  </button>
</DocumentFragment>
`;

exports[`Render loading state 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
    disabled=""
    type="button"
  >
    <span
      aria-hidden="true"
      class="spinner-border spinner-border-sm"
      role="status"
    />
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </button>
</DocumentFragment>
`;
